### DISCUSSION TITLE
link_to helper shows a tags as string

### DISCUSSION THREAD

**Discussion by merongivian** [yippee-fun/phlex#247] [Help] [answered]
I've read rails helpers are not completely integrated, is this something that is not yet implemented?

**Comment by joeldrapper** [comment]
That's right, we haven't adapted the `link_to` helper yet. I’m not sure if it’s worth doing since you can just write something like this.

You could always define your own adapter

```ruby
def link_to(...)
  raw helpers.link_to(...)
end
```

Or use `url_for` directly which will allow you to pass the content of the link as a block and also allow to use normal Phlex attributes on the link.

```ruby
def link_to(something, **attributes, &block)
  a(**attributes, href: helpers.url_for(something), &block)
end
```

Now you can use it like this.

```ruby
link_to(@article, class: "bla") { "Hello" }
```

The advantage of being able to pass a block is you can continue to use Phlex to put HTML inside the link.

```ruby
link_to @article do
  h1 { "Inside link" }
end
```

I expect we will ship an adapted `link_to` helper with this interface. ^^

**Comment by joeldrapper** [comment, marked as answer]
@merongivian 👋 hey, coming back to your question about `link_to`, `phlex-rails` now supports all Rails helpers via adapters which you can include as needed. To use `link_to`, just include `Phlex::Rails::Helpers::LinkTo` and it should work as expected.
